File: backend/app/models/schemas/document.py
from typing import Optional, Dict
from pydantic import BaseModel
from datetime import datetime

class DocumentBase(BaseModel):
    filename: Optional[str] = None
    url: Optional[str] = None
    description: Optional[str] = None

class DocumentCreate(DocumentBase):
    user_id: int

class DocumentAnalysis(BaseModel):
    invoiceNumber: Optional[str] = None
    clientName: Optional[str] = None
    invoiceDate: Optional[str] = None
    dueDate: Optional[str] = None
    totalAmount: Optional[str] = None
    classification: Optional[str] = None
    fraudCheck: Optional[Dict[str, str]] = None
    insights: Optional[str] = None

class DocumentResponse(DocumentBase):
    id: int
    created_at: datetime
    file_url: Optional[str] = None
    file_type: Optional[str] = "PDF"
    file_size: Optional[int] = 0
    page_count: Optional[int] = 1
    status: Optional[str] = "processed"
    analysis: Optional[DocumentAnalysis] = None
    
    class Config:
        from_attributes = True

class IntegrationResponse(BaseModel):
    document_id: int
    integrated_with: str
    success: bool
    reference_id: Optional[str] = None
    details: str 

File: frontend/src/types/document.ts
export type DocStatus = "uploaded" | "processing" | "failed";

export interface Document {
  id: string;
  filename: string;
  created_at: string;
  file_size: number;
  file_type: string;
  status: DocStatus;
  description?: string;
}

// Document Types
export interface DocumentData {
  id: string;
  filename: string;
  pageCount: number;
  currentPage: number;
  created_at: string;
  file_type: string;
  file_size: number;
  status: string;
  file_url?: string;
  analysis?: DocumentAnalysis;
  description?: string;
}

// Analysis Types
export interface DocumentAnalysis {
  invoiceNumber?: string;
  clientName?: string;
  invoiceDate?: string;
  dueDate?: string;
  totalAmount?: string;
  classification?: string;
  fraudCheck?: {
    status: string;
    details: string;
  };
  insights?: string;
}

// API Response Types
export interface DocumentApiResponse {
  id: string;
  filename: string;
  page_count?: number;
  created_at: string;
  file_size?: number;
  file_type?: string;
  status?: string;
  file_url?: string;
  analysis?: DocumentAnalysis;
  description?: string;
}
